Flow.Launcher/ViewModel/ResultsViewModel.cs

```diff
@@ -139,39 +139,9 @@ public void KeepResultsExcept(PluginMetadata metadata)
         /// </summary>
         public void AddResults(List<Result> newRawResults, string resultId)
         {
-            lock (_collectionLock)
-            {
-                var newResults = NewResults(newRawResults, resultId);
-
-                // https://social.msdn.microsoft.com/Forums/vstudio/en-US/5ff71969-f183-4744-909d-50f7cd414954/binding-a-tabcontrols-selectedindex-not-working?forum=wpf
-                // fix selected index flow
-                var updateTask = Task.Run(() =>
-                {
-                    // update UI in one run, so it can avoid UI flickering
-
-                    Results.Update(newResults);
-                    if (Results.Any())
-                        SelectedItem = Results[0];
-                });
-                if (!updateTask.Wait(300))
-                {
-                    updateTask.Dispose();
-                    throw new TimeoutException("Update result use too much time.");
-                }
-
-            }
+            var newResults = NewResults(newRawResults, resultId);
 
-            if (Visbility != Visibility.Visible && Results.Count > 0)
-            {
-                Margin = new Thickness { Top = 8 };
-                SelectedIndex = 0;
-                Visbility = Visibility.Visible;
-            }
-            else
-            {
-                Margin = new Thickness { Top = 0 };
-                Visbility = Visibility.Collapsed;
-            }
+            UpdateResults(newResults);
         }
         /// <summary>
         /// To avoid deadlock, this method should not called from main thread
```

Comment:
@taooceros do you know if this comment is still relevant?

Reply:
I think yes, because only results from main result set will be added via resultUpdateQueue, the one for context menu and history is still added directly.
However, I am not sure the deadlock meaning here. Maybe just avoiding potential stunt of main ui due to lock machanism.

Reply:
cool, thank you